tests/examples/shims/screen.lua

```diff
@@ -85,8 +87,36 @@ function screen._get_extents()
     return xmax.geometry.x+xmax.geometry.width, ymax.geometry.y+ymax.geometry.height
 end
 
+-- The way the `screen` module is used to store both number and object is
+-- problematic since it can cause invalid screens to be "resurrected".
+local function catch_invalid(_, s)
+    -- The CAPI implementation allows `nil`
+    if s == nil or type(s) == "string" then return end
+
+    assert(screen ~= s, "Some code tried (and failed) to shadow the global `screen`")
+
+    -- Valid numbers wont get this far.
```

Comment:
Why not? Only `nil` and `string` were checked so far.

Reply:
Because if you had 42 screens, remove them all, then add 2 screens, if you `screen[42]`, your code probably cached the screen number and that's evil (but common for older code written with v3.5 in mind).